connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -488,6 +488,14 @@ class SparkSession private[sql] (
   @scala.annotation.varargs
   def addArtifacts(uri: URI*): Unit = client.addArtifacts(uri)
 
+  /**
+   * Register a [[ClassFinder]] for dynamically generated classes.
+   *
+   * @since 3.4.0
+   */
+  @Experimental
```

Comment:
There is part of me that thinks we should be doing this as part of building the session, instead of adding it while running.

Reply:
> There is part of me that thinks we should be doing this as part of building the session, instead of adding it while running.

I tried that initially but the `repl.sess` that we pass over in `spark.registerClassFinder(new AmmoniteClassFinder(repl.sess))` is only available once the REPL is instantiated

Reply:
> Also mark @Private?
Wrt the above comment, we can't since we register the class finder as part of the predef code. 
I'm not too well versed with Ammonite internals, I wonder if we can get the REPL session without having to go through predef code.